app/services/preview_asset_service.rb

```diff
@@ -0,0 +1,39 @@
+# frozen_string_literal: true
+
+class PreviewAssetService
+  attr_reader :edition
+
+  def initialize(edition)
+    @edition = edition
+  end
+
+  def upload_assets
+    edition.image_revisions.each do |image_revision|
+      image_revision.ensure_assets
+      image_revision.assets.each { |asset| upload_asset(asset) }
+    end
+
+    edition.file_attachment_revisions.each do |file_attachment_revision|
+      file_attachment_revision.ensure_assets
+      file_attachment_revision.assets.each { |asset| upload_asset(asset) }
+    end
+  end
+
+  def upload_asset(asset)
+    return unless asset.absent?
+
+    auth_bypass_id = EditionUrl.new(edition).auth_bypass_id
+    file_url = AssetManagerService.new.upload(asset, auth_bypass_id)
+    asset.update!(file_url: file_url, state: :draft)
+  rescue GdsApi::BaseError => e
+    GovukError.notify(e)
+    raise
+  end
+
+  def pending?(asset)
```

Comment:
minor consistency issue that the method above this one uses asset in it's name whereas this one has it implied by argument.

Reply:
Good point - I've renamed this one to 'can_preview_asset?' and inverted the code where it was used; pending was a bit vague to be useful at this point, I think. Doing this also made me spot I'd missed the unit tests for this method :-).

Reply:
Heh I was going to suggest changing pending but I noticed it was used liberally in the trello card and that the Asset Manager alternative was "accessible" which was of course rather ambiguous.